src/js/libgeo/GeoOps.js

```diff
@@ -1725,12 +1735,68 @@ geoOps.TrMoebiusP.kind = "P";
 geoOps.TrMoebiusP.updatePosition = function(el) {
     var t = csgeo.csnames[(el.args[0])];
     var p = csgeo.csnames[(el.args[1])].homog;
-    var l1 = List.productMV(t.mat1, p);
-    var l2 = List.productMV(t.mat2, p);
-    el.homog = List.normalizeMax(List.cross(l1, l2));
+
+    el.homog = geoOps._helper.TrMoebiusP(p, t);
     el.homog = General.withUsage(el.homog, "Point");
 };
 
+geoOps._helper.TrMoebiusP = function(p, Tr) {
+    var l1 = List.productMV(Tr.mat1, p);
+    var l2 = List.productMV(Tr.mat2, p);
+    return List.normalizeMax(List.cross(l1, l2));
+};
+
+geoOps._helper.TrMoebiusPoints = function(tr, a1, a2, a3) {
+    var b1 = geoOps._helper.TrMoebiusP(a1, tr);
+    var b2 = geoOps._helper.TrMoebiusP(a2, tr);
+    var b3 = geoOps._helper.TrMoebiusP(a3, tr);
+    return [b1, b2, b3];
+};
+
+geoOps.TrMoebiusL = {};
+geoOps.TrMoebiusL.kind = "C";
+geoOps.TrMoebiusL.updatePosition = function(el) {
+    var t = csgeo.csnames[(el.args[0])];
+    var l = csgeo.csnames[(el.args[1])].homog;
+
+    var getRandLine = function() {
+        var rline = List.realVector([Math.random() - 0.5, Math.random() - 0.5, Math.random() - 0.5]);
+        return List.normalizeMax(rline);
+    };
+
+    var a1 = List.cross(getRandLine(), l);
+    var a2 = List.cross(getRandLine(), l);
+    var a3 = List.cross(getRandLine(), l);
+
+    var Pts = geoOps._helper.TrMoebiusPoints(t, a1, a2, a3);
+
+    el.matrix = List.normalizeMax(geoOps._helper.ConicBy5(null, Pts[0], Pts[1], Pts[2], List.ii, List.jj));
+    el.matrix = General.withUsage(el.matrix, "Circle");
+};
+
+geoOps.TrMoebiusArc = {};
+geoOps.TrMoebiusArc.kind = "C";
+geoOps.TrMoebiusArc.updatePosition = function(el) {
+    var t = csgeo.csnames[(el.args[0])];
+    var Arc = csgeo.csnames[(el.args[1])];
+
+    var a1 = Arc.startPoint;
+    var a2 = Arc.viaPoint;
+    var a3 = Arc.endPoint;
+
+    var Pts = geoOps._helper.TrMoebiusPoints(t, a1, a2, a3);
+    var b1 = Pts[0],
+        b2 = Pts[1],
+        b3 = Pts[2];
```

Comment:
You're right, i just wanted to avoid code doubling. Sometimes i really miss calling function by reference (without the object detour). I will upload a fix in a minute.